File: js/tests/spaceBetweenCharacters.test.js
const spaceMeOut = require("../spaceBetweenCharacters");

const data = [
  ["space", "s p a c e"],
  ["far out", "f a r   o u t"],
  ["elongated musk", "e l o n g a t e d   m u s k"],
  ["long", "l o n g"],
  ["123", "1 2 3"],
  ["a1b2c3", "a 1 b 2 c 3"]
];

describe("spaceMeOut", () => {
  describe("Return string with spaces between characters", () => {
    test.each(data)("spaceMeOut(%p) = %p", (str, result) => {
      let actual = spaceMeOut(str);

      expect(actual).toEqual(result);
    });
  });
});

File: CSharp/Tests/SpaceBetweenCharactersTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class SpaceBetweenCharactersTest
    {
        [Theory]
        [InlineData("space", "s p a c e")]
        [InlineData("far out", "f a r   o u t")]
        [InlineData("elongated musk", "e l o n g a t e d   m u s k")]
        [InlineData("long", "l o n g")]
        [InlineData("123", "1 2 3")]
        [InlineData("a1b2c3", "a 1 b 2 c 3")]
        public void SpaceMeOut_StringInputValue_ReturnStringWithSpacesBetweenCharacters(string str, string expected)
        {
            var actual = SpaceBetweenCharacters.SpaceMeOut(str);

            Assert.Equal(expected, actual);
        }
    }
}